### PR TITLE
Show page

### PR DESCRIPTION
**Pull request by cookiecuttershark**
Created a show page for each fashion item. Displays a form to add a review, and reviews for each fashion item.

### CODE REVIEW

File: src/main/frontend/app/Show.js [hunk 1]
```diff
@@ -1,35 +1,97 @@
...
+      })
+
+      fetch("api/v1/user")
+      .then(resp => {
+        return resp.json()
+      })
+      .then(body => {
+        this.setState({username:body})
+      })
+  }
```
Comment (prophetsfall) [inline src/main/frontend/app/Show.js]: I think there is a better way to write this. I'm going to do some research and might ask you to refactor later. But I won't hold the merge up for it

File: src/main/frontend/app/containers/ReviewContainer.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    render() {
+        let bodyTypeArray = ['straight','pear','oval', 'diamond']
+        let bodyTypeOptions = bodyTypeArray.map((name, i) => {
+            return <option key={i} value={i}>{name}</option>
+        })
+        let sizeArray = ['US 0', 'US 2', 'US 4', 'US 8', 'US 10', 'US 12', 'US 14', 'US 16', 'US 18', 'US 20']
+        let sizeOptions = sizeArray.map((name, i) => {
+            return <option key={i} value={i}>{name}</option>
+        })
```
Comment (prophetsfall) [inline src/main/frontend/app/containers/ReviewContainer.js]: It's probably worth abstracting bodyTypeArray and sizeArray to their own constants.

File: src/main/java/com/launchacademy/giantleap/models/FashionItemReview.java [hunk 3]
```diff
@@ -38,11 +39,27 @@
...
     private String comment;
 
-    @URL
     @Column
     private String photo;
 
     @Column
     private Integer rating;
 
+<<<<<<< HEAD
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/FashionItemReview.java]: You still have the remnants of a merge conflict here. 
You need to ensure you remove the <<<<<<< HEAD, the =====, and the >>>>>> (and hash)

File: src/main/java/com/launchacademy/giantleap/repositories/FashionItemRepository.java [hunk 4]
```diff
@@ -4,5 +4,9 @@
 import org.springframework.data.repository.PagingAndSortingRepository;
 
 public interface FashionItemRepository extends PagingAndSortingRepository<FashionItem, Integer> {
+<<<<<<< HEAD
+  FashionItem findById(int id);
+=======
   FashionItem findByName(String name);
+>>>>>>> 9b282cc261d648f8c0090a24a0ab8d2468b25dbf
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/repositories/FashionItemRepository.java]: Merge conflicts remnant here too

File: src/main/frontend/app/Show.js (hunk 1 above)
Comment (cookiecuttershark) [reply src/main/frontend/app/Show.js]: I agree. I was testing it and meant to delete that.

Comment (prophetsfall) [reply src/main/frontend/app/Show.js]: I did some research (and Hieu helped actualize it) and it looks like what we want to do here is combine the JSON we're fetching on the backend to return an object which contains both the items and the reviews. 

I can follow up with you more in person.

File: src/main/frontend/app/components/TextArea.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  return(
+    <div>
+      <label>{props.label}</label>
+      <textarea name={props.name} onChange={props.handleChange} value={props.value}>
+      </textarea>
+    </div>
+  )
+}
+
+export default TextField
```
Comment (prophetsfall) [inline src/main/frontend/app/components/TextArea.js]: Looks like this should be called TextArea in the const and export.

File: src/main/java/com/launchacademy/giantleap/models/FashionItem.java [hunk 6]
```diff
@@ -53,6 +56,7 @@
   @Column(name="item_size")
   private Integer itemSize;
 
+
```
Comment (prophetsfall) [inline src/main/java/com/launchacademy/giantleap/models/FashionItem.java]: please remove extra line

**Pull request merged**